tracing/src/main/java/com/palantir/remoting3/tracing/CloseableTracer.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting3.tracing;
+
+import com.palantir.remoting.api.tracing.SpanType;
+
+/**
+ * Wraps the {@link Tracer} methods in a closeable resource to enable the usage of the try-with-resources pattern.
+ *
+ * Usage:
+ *   try (CloseableTracer trace = CloseableTracer.start("traceName")) {
+ *       [...]
+ *   }
+ */
+public final class CloseableTracer implements AutoCloseable {
```

Comment:
maybe just have this be an enum?

Reply:
What is the advantage?

Reply:
Thinking about it some more, I'm not 100% sure on what you mean. Do you mean something like:
```
enum CloseableTracer implements AutoCloseable {
  INSTANCE;
  static CloseableTracer startSpan(...) { ... }
}
```
?

In that case what would stop the user from accessing `CloseableTracer.INSTANCE` directly? It's also not obvious to me what the advantage would be. Did you mean something else?

Reply:
oh, sorry. So I was just suggesting that given that you only want there to be one of these, one thing which people frequently end up doing is making it an enum. But I don't really care and don't want to block this.